azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -1015,31 +1017,37 @@ public void uploadFlowFile(final int projectId, final int projectVersion, final
 
   @Override
   public File getUploadedFlowFile(final int projectId, final int projectVersion,
-      final int flowVersion, final String flowName) throws ProjectManagerException {
+      final String flowFileName,
+      final int flowVersion) throws ProjectManagerException, IOException {
     final FlowFileResultHandler handler = new FlowFileResultHandler();
 
     final List<byte[]> data;
-    // Todo jamiesjc: delete the flow file after used.
-    final File file = new File(this.tempDir, flowName);
+    final File tempDir = Utils.createTempDir(this.tempDir);
+    final File file = new File(tempDir, flowFileName);
     try (final FileOutputStream output = new FileOutputStream(file);
         final BufferedOutputStream bufferedStream = new BufferedOutputStream(output)) {
       try {
         data = this.dbOperator
             .query(FlowFileResultHandler.SELECT_FLOW_FILE, handler,
-                projectId, projectVersion, flowName, flowVersion);
+                projectId, projectVersion, flowFileName, flowVersion);
       } catch (final SQLException e) {
         logger.error(e);
-        throw new ProjectManagerException("Failed to query uploaded flow file " + flowName + ".",
-            e);
+        if (tempDir.exists()) {
+          FileUtils.deleteDirectory(tempDir);
+        }
+        throw new ProjectManagerException(
+            "Failed to query uploaded flow file " + flowFileName + ".", e);
       }
 
-      try {
+      if (data != null && !data.isEmpty()) {
```

Comment:
can yaml parser handle empty file?

Reply:
Yaml parser can handle empty file. But we still need to explicitly check the parsed result is not null.